sql-statements/sql-statement-create-index.md

```diff
@@ -232,6 +232,119 @@ Therefore, when the query performance outweighs the insert and update performanc
 
 Expression indexes have the same syntax and limitations as in MySQL. They are implemented by creating indexes on generated virtual columns that are invisible, so the supported expressions inherit all [limitations of virtual generated columns](/generated-columns.md#limitations).
 
+## Multi-valued index
+
+> **Warning:**
+>
+> For the current version, this feature is still experimental and not recommended for production environments.
+
+Multi-valued index is a kind of secondary index defined on an array column. In an ordinary index, one index record corresponds to one data record (1:1). In a multi-valued index, multiple index records correspond to one data record (N:1). Multi-valued indexes are used to index JSON arrays. For example, a multi-valued index defined on the `zipcode` field will generate one index record for each element in the `zipcode` array.
+
+```json
+{
+    "user":"Bob",
+    "user_id":31,
+    "zipcode":[94477,94536]
+}
+```
+
+### Create a multi-valued index
+
+You can create a multi-valued index by using the `CAST(... AS ... ARRAY)` expression in the index definition, as creating an expression index.
+
+```sql
+CREATE TABLE customers (
+    id BIGINT NOT NULL AUTO_INCREMENT PRIMARY KEY,
+    name CHAR(10),
+    custinfo JSON,
+    INDEX zips((CAST(custinfo->'$.zipcode' AS UNSIGNED ARRAY)))
+);
+```
+
+You can define a multi-valued index as a unique index.
+
+```sql
+CREATE TABLE customers (
+    id BIGINT NOT NULL AUTO_INCREMENT PRIMARY KEY,
+    name CHAR(10),
+    custinfo JSON,
+    UNIQUE INDEX zips( (CAST(custinfo->'$.zipcode' AS UNSIGNED ARRAY)))
+);
+```
+
+When a multi-valued index is defined as a unique index, an error is reported if you try to insert duplicate data.
+
+```sql
+mysql> INSERT INTO customers VALUES (1, 'pingcap', '{"zipcode": [1,2]}');
+Query OK, 1 row affected (0.01 sec)
+mysql> INSERT INTO customers VALUES (1, 'pingcap', '{"zipcode": [2,3]}');
+ERROR 1062 (23000): Duplicate entry '2' for key 'customers.zips'
+```
+
+The same record can have duplicate values, but when different records have duplicate values, an error is reported.
+
+```sql
+-- Insert successfully
+INSERT INTO t1 VALUES('[1,1,2]');
+INSERT INTO t1 VALUES('[3,3,3,4,4,4]');
+
+-- Insert failed
+INSERT INTO t1 VALUES('[1,2]');
+INSERT INTO t1 VALUES('[2,3]');
+```
+
+You can also define a multi-valued index as a compound index:
```

Comment:
```suggestion
You can also define a multi-valued index as a composite index:
```
`composite index` is used in our existing docs: 
<img width="591" alt="image" src="https://user-images.githubusercontent.com/59654584/215961756-0fc12237-a320-430d-bf6e-d39934c2225f.png">